buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -173,21 +173,14 @@ public static void configureCompile(Project project) {
                 compileOptions.setEncoding("UTF-8");
                 compileOptions.setIncremental(true);
 
-                // TODO: use native Gradle support for --release when available (cf. https://github.com/gradle/gradle/issues/2510)
                 final JavaVersion targetCompatibilityVersion = JavaVersion.toVersion(compileTask.getTargetCompatibility());
-                compilerArgs.add("--release");
-                compilerArgs.add(targetCompatibilityVersion.getMajorVersion());
-
+                compileOptions.getRelease().set(Integer.parseInt(targetCompatibilityVersion.getMajorVersion()));
```

Comment:
Should we use a `Provider` here so we an avoid any future issues of configuration ordering if individual modules override target compatibility version?

Reply:
strictly speaking not required as this is done in an afterEvaluate block, BUT it brings use closer to getting rid of that   `afterEvaluate` . On it.